### ISSUE TITLE
[Feature Request] For artist's album Release-Type lists, please change "Unknown" to "Album"

### ISSUE THREAD

**Issue by troycarpenter** [mitchray/ample#31]
For a listing of an artist's albums, if there is no release-type defined, the heading for the section is "UNKNOWN"
![image](https://user-images.githubusercontent.com/8419000/182910541-01af4935-5cc4-4576-975d-0763a0031437.png)

In Ampache, the assumption is that in absence of the release-type, the type is "ALBUM"
![image](https://user-images.githubusercontent.com/8419000/182910782-97f204ab-8401-408b-b993-03ac52cfaa5f.png)

Please change the "UNKNOWN" heading to "ALBUM" for those albums that do not include a release-type.

**Comment by troycarpenter** [comment]
Also, I'm not sure but I think these are presented alphabetically.  If not, I would like to have the "ALBUM" or "UNKNOWN" section listed first, since for most artists that's where the majority of their albums will be.

**Comment by mitchray** [comment]
The order is taken from Ampache's setting "Album - Group per release type sort" under Preferences >Interface > Library, so having "album,unknown,etc" will have them both be at the top

**Comment by mitchray** [comment]
I can see why Ampache would default to "album" for missing values, to cater to a wider audience who might not have properly tagged files

I would prefer to keep them separate to indicate the missing value however

**Comment by Jieiku** [comment]
I also would prefer to keep them separate, that way should I ever notice it I can go back and fix the tags.

**Comment by troycarpenter** [comment]
Yikes.  Well, I have over 1800 albums, so I'm not planning to tag those.  Where does Ampache even pull the release type field from?  Is there an mp3 tag for that?

**Comment by Jieiku** [comment]
I am not sure if it is an mp3 tag or not, but it seems when you add a catalog to Ampache you can use %r in the import pattern.

So I would say it is information that can be retrieved from Ampache's database, but no idea if Ampache can get this from tags as wel as the import pattern.

Let me know if you find out if there is a specific mp3 tag for this.

I just noticed that ALL of my library shows Uknown under ample, I honestly had not even noticed it.

**Comment by troycarpenter** [comment]
When I have an album that I need to add the release type, I edit the album one at a time in Ampache.  So all this time I thought the default type was album, but I guess they were empty.  I was only hand changing the release type for non-album releases.

**Comment by mitchray** [comment]
[It depends on the file type](https://picard-docs.musicbrainz.org/en/appendices/tag_mapping.html#id33) but it appears MP3 is **TXXX:MusicBrainz Album Type**

If you use something like MusicBrainz Picard (definitely) or Foobar/MP3Tag (presumably) it will handle the correct type for your file

Or you can disable the grouping in Ampache altogether by disabling "Album - Group per release type"

**Comment by troycarpenter** [comment]
> I am not sure if it is an mp3 tag or not, but it seems when you add a catalog to Ampache you can use %r in the import pattern.
> ![2022-08-04_20-15-19](https://user-images.githubusercontent.com/106644/182993824-ba977b07-ae13-4361-9809-f35175c633dd.png)
> 
> 

That implies the release type is encoded in the track filename and can be used to inport into the database, not that there's an mp3 field for it.

**Comment by troycarpenter** [comment]
> [It depends on the file type](https://picard-docs.musicbrainz.org/en/appendices/tag_mapping.html#id33) but it appears MP3 is **TXXX:MusicBrainz Album Type**
> 
> If you use something like MusicBrainz Picard (definitely) or Foobar/MP3Tag (presumably) it will handle the correct type for your file
> 
> Or you can disable the grouping in Ampache altogether by disabling "Album - Group per release type"
> 

Good to know about that tag.

And the grouping is useful to keep compilations and other release types out of the regular album releases list.

**Comment by Jieiku** [comment]
> That implies the release type is encoded in the track filename and can be used to inport into the database, not that there's an mp3 field for it.

OR that the release type is part of the directory structure.

**Comment by Jieiku** [comment]
I just disabled "Album - Group per release type" but it still shows UNKNOWN

**Comment by troycarpenter** [comment]
> [It depends on the file type](https://picard-docs.musicbrainz.org/en/appendices/tag_mapping.html#id33) but it appears MP3 is **TXXX:MusicBrainz Album Type**
> 
> If you use something like MusicBrainz Picard (definitely) or Foobar/MP3Tag (presumably) it will handle the correct type for your file
> 
> Or you can disable the grouping in Ampache altogether by disabling "Album - Group per release type"

I found a Compilation album that was showing up on an artist page.  I added the tag "MUSICBRAINZ_ALBUMTYPE" set to "Compilation" to the songs and updated the tags in Ampache.  The album now shows up under the Compilation section of the artist's album listing.

I don't normally embed the Musicbrainz tags, but I'll look into it.  30k songs to retag yet again.  Seems like my life is all about retagging music files.

**Comment by Jieiku** [comment]
Thank you for sharing the tag name, I can probably work that into my bash script that handles my tagging.

**Comment by troycarpenter** [comment]
I'm trying Picard which pulled in lots of extra tags (most I don't really care about).  What I didn't like is that it messed with the track numbers and disc numbers.  I need to see if there's a way to protect those fields.

**Comment by Jieiku** [comment]
troy, I do the same thing in my filenames, and then my bash script properly sets the disc number and track number in the actual tags:
![2022-08-05_08-20-04](https://user-images.githubusercontent.com/106644/183108624-630a0319-0ade-4610-bee1-257d8d3b60e4.png)

**Comment by troycarpenter** [comment]
I've started down the rabbit hole of Musicbrainz tagging.  I'm about half done (mostly the mainstream music).  I've found my taste in music is a bit off the beaten path and I've had to submit not a few albums to existing artists, as well as create new artists and add their albums.  Picard makes that rather painless, though.  I also don't agree with how some of the albums / tracks are tagged.  I think some people submitted entries in the past before understanding how they should be entered (or maybe it's a preference thing).

What I'm now dealing with are all the cross-artist tags that didn't exist before.  It's nice to have links between guest artists and such, but it does change slightly how I interface with my music.  Not worse, just different and maybe better in finding other tracks on which an artist appears.  It's also interesting to see how Ampache uses the MB tags vs. MP3 tags in database and display choices.  It mostly leans on MP3 tags for display purposes, while MB tags create database relationships between artists.
